src/Monarch/Document.ts

```diff
@@ -14,47 +14,88 @@ import { OutputHandlersList } from 'monarch/Monarch/VirtualDom/OutputHandlersLis
 import { VirtualDomTree } from 'monarch/Monarch/VirtualDom/VirtualDomTree'
 import { unsafe_uncurried_applyPatchTree } from 'monarch/Monarch/VirtualDom/PatchTree'
 import { unsafe_uncurried_mount } from 'monarch/Monarch/VirtualDom'
-import { DiffWorkEnvironment, DiffWork, DiffWorkResult, mkRootDiffWork, unsafe_uncurried_performDiffWork } from 'monarch/Monarch/VirtualDom/DiffWork'
+import {
+    DiffWorkEnvironment,
+    DiffWork,
+    DiffWorkResult,
+    mkRootDiffWork,
+    unsafe_uncurried_performDiffWork,
+} from 'monarch/Monarch/VirtualDom/DiffWork'
 import { mkScheduler } from 'monarch/Monarch/Scheduler'
 
 interface DispatchMessage<message> {
     (message: message): Effect<Unit>
 }
 
-interface Spec<input, model, message> {
+interface DispatchOutput<output> {
+    (output: output): Effect<Unit>
+}
+
+interface OutputCallback<output> {
+    (output: output): Effect<Unit>
+}
+
+interface Output<output> {
+    (callback: OutputCallback<output>): Effect<Unit>
+}
+
+// prettier-ignore
+interface Spec<input, model, message, output> {
     input: input
     init(input: input): model
     update: (message: message) => (model: model) => model
     view(model: model): VirtualDomTree<message>
     container: HTMLElement
-    runCommand: (message: message) => (model: model) => (dispatchMessage: DispatchMessage<message>) => Effect<Unit>
+    runCommand: (message: message) => (model: model) => (dispatchMessage: DispatchMessage<message>) => (dispatchOutput: DispatchOutput<output>) => Effect<Unit>
 }
 
-interface State<model, message> {
+interface State<model, message, output> {
     commitedVirtualDomTree: VirtualDomTree<message>
     diffWork?: DiffWork<any, any>
     diffWorkResult?: DiffWorkResult<message>
     hasRequestedAsyncRendering: boolean
     model: model
+    outputCallbacks: OutputCallback<output>[]
```

Comment:
There is no need for handling multiple output handlers in the `Monarch.Document`. How about this?

```purs
Monarch.mkDocument { ...
                   , onOutput: \output -> ...
                   }
```